### PR TITLE
Refactor testimonials section

### LINKED ISSUE #1575: [feature]: add testimonials section to blog website
### Is there an existing feature request for this?

- [X] I have searched the existing issues

### Summary

There are user testimonials of keploy users that can be posted on the blog website. You can pick content of testimonials from the main website, start with 3-4 cards. Please feel free to discuss design.

This component will also be re-used on the landing page as the curr one needs improvement. 

### Why should this be worked on?

Builds trust and validates claims with new users.

### Repository

blog-website

### PR DESCRIPTION
**Pull request by sahadat-sk**
Fixes: [#1575](https://github.com/keploy/keploy/issues/1575)

### CODE REVIEW

**Comment by sahadat-sk** [comment]
I have changed the testimonials section according to the design and added the animation, but I am not sure if it should be in a carousel or not.

image

**Review by Hermione2408 — CHANGES_REQUESTED** [review]
Hey @sahadat-sk 
1.The scroll trigger should trigger at bottom when section starts to show in the view right now when the whole section is loaded then it is showing text.
2.Please remove the background from the testimonial and follow the color scheme mentioned in README

**Comment by sahadat-sk** [comment]
> Hey @sahadat-sk 1.The scroll trigger should trigger at bottom when section starts to show in the view right now when the whole section is loaded then it is showing text. 2.Please remove the background from the testimonial and follow the color scheme mentioned in README

I have made the changes, can you please review them?

**Comment by Hermione2408** [comment]
> > Hey @sahadat-sk 1.The scroll trigger should trigger at bottom when section starts to show in the view right now when the whole section is loaded then it is showing text. 2.Please remove the background from the testimonial and follow the color scheme mentioned in README
> 
> I have made the changes, can you please review them?

Hey @sahadat-sk I think you misunderstood Please remove the background from the testimonial and after that for text you can follow keploy color scheme

File: components/testimonials.tsx [hunk 1]
```diff
@@ -29,137 +36,93 @@ const testimonialDataMobile = [
...
-      {/*        </div>*/}
-      {/*      </div>*/}
-      {/*    </div>*/}
+  useGSAP(
+    () => {
+      gsap.to('.quote-character', {
+        scrollTrigger: {
+          trigger: '.quote-container',
+          start: 'top bottom',
+          end: 'bottom+=500 bottom',
```
Comment (Hermione2408) [inline components/testimonials.tsx]: We can reduce the end of scroll trigger as right now even when the whole section is visible the text is opaque 
Screenshot from 2024-03-20 23-51-55

**Review by Hermione2408 — CHANGES_REQUESTED** [review]
Please try to avoid linting changes in the PR

**Comment by sahadat-sk** [comment]
> > > Hey @sahadat-sk 1.The scroll trigger should trigger at bottom when section starts to show in the view right now when the whole section is loaded then it is showing text. 2.Please remove the background from the testimonial and follow the color scheme mentioned in README
> > 
> > 
> > I have made the changes, can you please review them?
> 
> Hey @sahadat-sk I think you misunderstood Please remove the background from the testimonial and after that for text you can follow keploy color scheme

Sorry for the misunderstanding, can you review the changes now?

**Review by Hermione2408 — CHANGES_REQUESTED** [review]
Hey , You have used gsap , but you haven't pushed changes related to package.json and package-lock.json for installing dependencies

**Comment by sahadat-sk** [comment]
> Hey , You have used gsap , but you haven't pushed changes related to package.json and package-lock.json for installing dependencies

Done!

**Review by Hermione2408 — CHANGES_REQUESTED** [review]
Hey @sahadat-sk 
I've checked, and the Scroll Trigger functionality is working perfectly now, so great job on that! However, I've noticed that the testimonial section hasn't been updated to reflect Keploy's color scheme. Could you please adjust that?
Additionally, I've been thinking Instead of grouping them all together,we can move testimonial between different sections. For instance, we could place one testimonial after the "Keploy for Developers" section and another following the "Community" section.

**Comment by sahadat-sk** [comment]
> Hey @sahadat-sk I've checked, and the Scroll Trigger functionality is working perfectly now, so great job on that! However, I've noticed that the testimonial section hasn't been updated to reflect Keploy's color scheme. Could you please adjust that? Additionally, I've been thinking Instead of grouping them all together,we can move testimonial between different sections. For instance, we could place one testimonial after the "Keploy for Developers" section and another following the "Community" section.

Hey @Hermione2408, sorry for the confusion, I have been using the color scheme only (primary-200, specified in tailwind.config.js), can you please specify what exact color should I use for this purpose? thank you!

**Comment by Hermione2408** [comment]
> > Hey @sahadat-sk I've checked, and the Scroll Trigger functionality is working perfectly now, so great job on that! However, I've noticed that the testimonial section hasn't been updated to reflect Keploy's color scheme. Could you please adjust that? Additionally, I've been thinking Instead of grouping them all together,we can move testimonial between different sections. For instance, we could place one testimonial after the "Keploy for Developers" section and another following the "Community" section.
> 
> Hey @Hermione2408, sorry for the confusion, I have been using the color scheme only (primary-200, specified in tailwind.config.js), can you please specify what exact color should I use for this purpose? thank you!

Sure you can use text-accent-200 ,text-secondary-300

**Comment by sahadat-sk** [comment]
Hey @Hermione2408, I have made changes, Do let me know if it looks good or not

image

File: app/(default)/page.tsx [hunk 2]
```diff
@@ -1,30 +1,63 @@
...
+  description:
+    "Keploy is AI based test case and stubs/mocks generator for e2e testing. 90% test coverage in minutes with open source testing tool",
+};
 
-import Hero from '@/components/hero'
-import Features from '@/components/features'
-import ProblemBlocks from '@/components/problem-blocks'
-import Testimonials from '@/components/testimonials'
-import Newsletter from '@/components/newsletter'
+import Hero from "@/components/hero";
```
Comment (Hermione2408) [inline app/(default)/page.tsx]: Hey @sahadat-sk these are linting changes can you please revert them back?

File: app/(default)/page.tsx [hunk 3]
```diff
@@ -1,30 +1,63 @@
...
+    author: "Staff Engineer",
+    company: "@Nutanix",
+  },
+  // Add more testimonials as needed
+];
+
 export default function Home() {
   return (
     <>
-        <Hero />
```
Comment (Hermione2408) [inline app/(default)/page.tsx:19]: here too

**Review by Hermione2408 — CHANGES_REQUESTED** [review]
Hey @sahadat-sk, please revert the linting changes to avoid conflicts. Everything else looks good!

File: app/(default)/page.tsx (hunk 3 above)
Comment (sahadat-sk) [reply app/(default)/page.tsx:19]: I have reverted the linting changes, let me know if it is fine now.

**Comment by Hermione2408** [comment]
LGTM!  @nehagup

**Review by nehagup — COMMENTED** [review]
@sahadat-sk this looks very promising! Small change:

- can we please add a left to right orange container highlighting over company name and position, instead of static orange fading gradient?

**Comment by nehagup** [comment]
@sahadat-sk also please start this effect once the testimonial is height-10%, I showed this to a couple of folks, they couldn't notice this as off now.  :)

**Comment by sahadat-sk** [comment]
> @sahadat-sk this looks very promising! Small change:
> 
>     * can we please add a left to right orange container highlighting over company name and position, instead of static orange fading gradient?

Hey @nehagup, I have added an orange container with left-to-right animation instead of static gradient, can you review if it looks good?

**Review by nehagup — COMMENTED** [review]
@sahadat-sk glad that you got the idea! Can you please make it gradient, ending to transparent on the right most side? (refer to your screenshot with purple highlight)

Also, referring to my previous comment - it'll be great if the typewriting effect starts kicking in when the testimonial has come to vh-10% height and it should complete by mid of the screen(40-50% vh).

**Comment by sahadat-sk** [comment]
> @sahadat-sk glad that you got the idea! Can you please make it gradient, ending to transparent on the right most side? (refer to your screenshot with purple highlight)
> 
> Also, referring to my previous comment - it'll be great if the typewriting effect starts kicking in when the testimonial has come to vh-10% height and it should complete by mid of the screen(40-50% vh).

Hey @nehagup, I have changed the container background to gradient and also delayed the typewriting animation, can you review them now?

**Review by nehagup — APPROVED** [review]
LGTM! Thanks @sahadat-sk 👏

**Pull request merged**